### PR TITLE
Rearchitect @graknlabs_build_tools to @graknlabs_dependencies

### PR DESCRIPTION
**Pull request by lolski**
We have promoted build-tools to be the repositories that acts as a central point for dependencies, builders, and toolings. Hence, it will be renamed to `@graknlabs_dependencies`

The following changes are most noteworthy:
  - `//bin`: for storing `grakn`, `grakn.bat`, and `grakn-docker.sh` (moved from `@graknlabs_common`)
  - `//builder`: for things that you need during build time, such as Bazel rules (think `rules_nodejs`, `rules_proto`, etc).
  - `//distribution`: toolings related to assembling and deploying
  - `//image`: for VM and Docker images
  - `//library`: for dependencies that's actually used in the software such as Maven artifacts
  - `//tool`: for infrastructure-related toolings such as release automation and CI

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -14,30 +14,4 @@
...
-        }
-        """,
-)
-
-py_library(
-    name = "common",
-    srcs = ["common.py"],
-    visibility = ["//visibility:public"]
-)
+#
```
Comment (haikalpribadi) [inline BUILD:17]: new line

File: binary/BUILD [hunk 2]
```diff
@@ -15,5 +15,6 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
+#
 
-exports_files(["release_validate_deps.py"])
+exports_files(["grakn", "grakn.bat", "grakn-docker.sh"])
```
Comment (haikalpribadi) [inline binary/BUILD:20]: new line

File: binary/grakn.bat [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  echo Grakn Core Server is not included in this Grakn distribution^.
+  echo You may want to install Grakn Core Server or Grakn Core ^(all^)^.
+  goto exiterror
+)
+
+:exit
+exit /b 0
+
+:exiterror
+exit /b 1
```
Comment (haikalpribadi) [inline binary/grakn.bat:81]: new line

File: builder/bazel/deps.bzl [hunk 4]
```diff
@@ -40,27 +40,4 @@ def bazel_deps():
...
-    )
-
-def bazel_rules_python():
-    http_archive(
-        name = "rules_python",
-        url = "https://github.com/bazelbuild/rules_python/releases/download/0.0.2/rules_python-0.0.2.tar.gz",
-        strip_prefix = "rules_python-0.0.2",
-        sha256 = "b5668cde8bb6e3515057ef465a35ad712214962f0b3a314e551204266c7be90c",
-    )
+    )
```
Comment (haikalpribadi) [inline builder/bazel/deps.bzl:43]: new line

File: builder/java/deps.bzl [hunk 5]
```diff
@@ -0,0 +1,9 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive", "http_jar")
+
+def deps():
+    http_archive(
+        name = "rules_jvm_external",
+        strip_prefix = "rules_jvm_external-3.2",
+        sha256 = "82262ff4223c5fda6fb7ff8bd63db8131b51b413d26eb49e3131037e79e324af",
+        url = "https://github.com/bazelbuild/rules_jvm_external/archive/3.2.zip",
+    )
```
Comment (haikalpribadi) [inline builder/java/deps.bzl:9]: new line

File: distribution/BUILD [hunk 6]
```diff
@@ -15,3 +15,26 @@
...
+    visibility = ["//visibility:public"]
+)
+
+assemble_targz(
+    name = "assemble-bat-targz",
+    additional_files = {
+        "//binary:grakn.bat": 'grakn.bat',
+    },
+    visibility = ["//visibility:public"]
+)
```
Comment (haikalpribadi) [inline distribution/BUILD:40]: new line

File: distribution/apt/BUILD [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        "usr/local/bin/grakn": "/opt/grakn/core/grakn",
+        "opt/grakn/core/logs": "/var/log/grakn/",
+    },
+)
+
+deploy_apt(
+    name = "deploy-apt",
+    target = ":assemble-linux-apt",
+    deployment_properties = "//distribution:deployment.properties",
+)
```
Comment (haikalpribadi) [inline distribution/apt/BUILD:30]: new line

File: distribution/docker/deps.bzl [hunk 8]
```diff
@@ -0,0 +1,9 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive", "http_file", "http_jar")
+
+def deps():
+    http_archive(
+        name = "io_bazel_rules_docker",
+        sha256 = "14ac30773fdb393ddec90e158c9ec7ebb3f8a4fd533ec2abbfd8789ad81a284b",
+        strip_prefix = "rules_docker-0.12.1",
+        urls = ["https://github.com/bazelbuild/rules_docker/releases/download/v0.12.1/rules_docker-v0.12.1.tar.gz"],
+    )
```
Comment (haikalpribadi) [inline distribution/docker/deps.bzl:9]: new line

File: library/maven/artifacts.bzl [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    "org.hamcrest:hamcrest-all": "1.3",
+    "org.hamcrest:hamcrest-core": "1.3",
+    "org.hamcrest:hamcrest-library": "1.3",
+    "org.mockito:mockito-core": "2.6.4",
+    "org.slf4j:jcl-over-slf4j": "1.7.20",
+    "org.slf4j:log4j-over-slf4j": "1.7.20",
+    "org.slf4j:slf4j-api": "1.7.28",
+    "org.slf4j:slf4j-simple": "1.7.20",
+    "org.zeroturnaround:zt-exec": "1.10" # exclude: - "commons-io:commons-io"
+}
```
Comment (haikalpribadi) [inline library/maven/artifacts.bzl:35]: new line

File: tool/BUILD [hunk 10]
```diff
@@ -14,4 +14,10 @@
...
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
-#
\ No newline at end of file
+#
+
+py_library(
+    name = "common",
+    srcs = ["common.py"],
+    visibility = ["//visibility:public"]
+)
```
Comment (haikalpribadi) [inline tool/BUILD:23]: new line

Comment (haikalpribadi) [reply tool/BUILD:23]: I think this should sit in its own package

File: tool/bazelrun/BUILD [hunk 11]
```diff
@@ -0,0 +1,5 @@
+py_binary(
+    name = "rbe",
+    srcs = ["rbe.py"],
+    main = "rbe.py"
+)
```
Comment (haikalpribadi) [inline tool/bazelrun/BUILD:5]: new line

File: tool/bazelinstall/BUILD [hunk 12]
```diff
@@ -0,0 +1 @@
+exports_files([ "linux.sh", "mac.sh", "rbe.sh" ])
```
Comment (haikalpribadi) [inline tool/bazelinstall/BUILD:1]: new line

File: tool/cache/BUILD [hunk 13]
```diff
@@ -0,0 +1,5 @@
+py_binary(
+    name = "cache",
+    srcs = ["cache.py"],
+    main = "cache.py"
+)
```
Comment (haikalpribadi) [inline tool/cache/BUILD:5]: new line

File: tool/sync/BUILD [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+    deps = [
+        "//tool:common",
+        requirement("PyGithub"),
+        requirement("urllib3"),
+        requirement("chardet"),
+        requirement("idna"),
+        requirement("wrapt"),
+        requirement("certifi"),
+    ],
+)
```
Comment (haikalpribadi) [inline tool/sync/BUILD:16]: new line

File: tool/BUILD (hunk 10 above)
Comment (lolski) [reply tool/BUILD]: > new line

Solved in https://github.com/graknlabs/dependencies/pull/136/commits/e44a270ac97b4d28e56e409d91fd75663ccb164c

> I think this should sit in its own package

Solved in https://github.com/graknlabs/dependencies/pull/136/commits/7edfbf380aead527fb239423a2f51b3a6b0d6db8

File: BUILD (hunk 1 above)
Comment (lolski) [reply BUILD]: Solved in https://github.com/graknlabs/dependencies/pull/136/commits/e44a270ac97b4d28e56e409d91fd75663ccb164c

File: binary/BUILD (hunk 2 above)
Comment (lolski) [reply binary/BUILD]: Solved in https://github.com/graknlabs/dependencies/pull/136/commits/e44a270ac97b4d28e56e409d91fd75663ccb164c

**Pull request merged**